dali/operators/reader/loader/video_loader.cc

```diff
@@ -587,6 +588,21 @@ void VideoLoader::read_file() {
     auto frame = av_rescale_q(pkt->pts - file.start_time_,
                               file.stream_base_,
                               file.frame_base_);
+    if (seek_before_last_key) {
+      if (last_key_frame <= frame) {
+        LOG_LINE << "Failed to seek to one before the last keyframe: " << last_key_frame
+                 << " , arrived at "<< last_key_frame << ". Trying --last_key_frame\n";
+        --last_key_frame;
+        DALI_ENFORCE(frame != 0, "Rewind to the frame 0 and the decoding has't started, "
+                                 "please check if your video is not corrupted");
+        seek(file, last_key_frame - 1);
+        continue;
+      }
+      LOG_LINE << "Seek to one before the last keyframe: " << last_key_frame
+               << " , arrived at "<< frame << "\n";
+      seek_before_last_key = false;
+      last_key_frame = -1;
```

Comment:
Can we end up in a situation where we try to seek 1 before `-1`?

Reply:
Extended enforce if that is what you have been asking for.